### ISSUE TITLE
Decouple Storybook publishing from test & lint job in CI

### ISSUE THREAD

**Issue by lmorchard** [mozilla/experimenter#3894]
Currently, we do not reach the Storybook publishing step in CI if tests or linting fail. This should be separated for two reasons:

1. It would be nice to run the storybook publishing in parallel, and
2. It would be nice to have a published storybook for a PR even if it hasn't yet got passing tests or complete test coverage

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-548)
┆Issue Number: EXP-548

**Comment by jaredlockhart** [comment]
@lmorchard You read my mind!! I was going to file this exact ticket <3
